src/test/java/io/github/christophermanahan/carnitas/HTTPServerTest.java

```diff
@@ -20,23 +22,27 @@
     @BeforeEach
     void setup() {
         sent = new ArrayList<>();
+        request = "GET http://localhost:80/simple_get HTTP/1.1";
         logger = new TestLogger();
     }
 
     @Test
-    void sendsHTTPResponseAfterReceivingData() {
-        connection = new TestConnection("GET simple_get HTTP/1.1", sent);
+    void sendsHTTPResponsesWhileReceivingData() {
+        received = List.of(request, request, request);
```

Comment:
I, personally, find it hard to read when `request` and `sent` are in `setup()`. Especially when the tests become longer it can be hard to scroll back up to remember what's going on in the test. I feel like `setup` works nicely with initialization ex: `TestLogger`

I defer to Damon on which approach he would like you to use.